firebase-installations/src/main/java/com/google/firebase/installations/local/PersistedInstallation.java

```diff
@@ -124,6 +127,7 @@ private JSONObject readJSONFromFile() {
       }
       return new JSONObject(baos.toString());
     } catch (IOException | JSONException e) {
+      Log.e(TAG, "error encountered while reading persisted firebase installation." + e);
```

Comment:
String.format(Error while reading from disc (%s): %s", filename, exception)

If there is always only 1 file, you can omit the filename.

Reply:
I recommend against logging the exception as a string and instead passing it in separately:
`Log.e(msg, exc)`